src/sentry/workflow_engine/endpoints/organization_detector_open_periods.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import annotations
+
+from rest_framework.exceptions import ParseError
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases import OrganizationDetectorPermission, OrganizationEndpoint
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.paginator import GenericOffsetPaginator
+from sentry.api.utils import get_date_range_from_params
+from sentry.exceptions import InvalidParams
+from sentry.models.groupopenperiod import OpenPeriod, get_open_periods_for_group
+from sentry.models.organization import Organization
+from sentry.workflow_engine.models import Detector
+from sentry.workflow_engine.models.detector_group import DetectorGroup
+
+
+@region_silo_endpoint
+class OrganizationDetectorOpenPeriodsEndpoint(OrganizationEndpoint):
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
+    }
+    owner = ApiOwner.ISSUES
+
+    permission_classes = (OrganizationDetectorPermission,)
+
+    def convert_args(self, request: Request, detector_id, *args, **kwargs):
+        args, kwargs = super().convert_args(request, *args, **kwargs)
+        try:
+            detector = Detector.objects.select_related("project").get(id=detector_id)
+            if detector.project.organization_id != kwargs["organization"].id:
+                raise ResourceDoesNotExist
```

Comment:
do we usually check this way or do we use the database? as in `project__organization_id=`